### PR TITLE
fix: Default Comment on Policy Violation #2

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at e9f95b7</samp>

Add default comment for mileage policy violations. This change prevents sending invalid comments to the backend and enables users to override policy violations by explaining their reasons.

### <samp>🤖 Generated by Copilot at e9f95b7</samp>

> _`continueWithTransaction`_
> _Needs a comment for violation_
> _Default added now_

### <samp>🤖 Generated by Copilot at e9f95b7</samp>

*  Add a default comment for policy violations ([link](https://github.com/fylein/fyle-mobile-app/pull/2274/files?diff=unified&w=0#diff-04c216da6200af711aa0f53bcd7e253086762b5d123b0729fac3b1862ec5fc3eR2477-R2479))

## Clickup
https://app.clickup.com/t/85ztcza5a

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 1]
```diff
@@ -3923,7 +3923,7 @@ export class AddEditExpensePage implements OnInit {
       switchMap(() => this.continueWithPolicyViolations(err.policyViolations, err.policyAction)),
       switchMap((continueWithTransaction: { comment: string }) => {
         if (continueWithTransaction) {
-          if (continueWithTransaction.comment === '') {
+          if (continueWithTransaction.comment === '' || continueWithTransaction.comment === null) {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:3926]: see if this works:
`continueWithTransaction.comment ?? '' === ''`

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 2]
```diff
@@ -2474,6 +2474,9 @@ export class AddEditMileagePage implements OnInit {
     return from(this.continueWithPolicyViolations(err.policyViolations, err.policyAction)).pipe(
       switchMap((continueWithTransaction: { comment: string }) => {
         if (continueWithTransaction) {
+          if (continueWithTransaction.comment === '' || continueWithTransaction.comment === null) {
```
Comment (Dimple16) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:2477]: same for all

**Review by Dimple16 — APPROVED** [review]
one comment

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.ts:3926]: This did not work, no default comment was added

**Review by Julias0 — APPROVED** [review]

**Pull request merged**